### PR TITLE
incrementally applying role checking to UI and API

### PR DESCRIPTION
**Pull request by 7ep**

### CODE REVIEW

**Review by coveros-matt-taylor — APPROVED** [review]
Gorgeous. Amazing

**Review by 7ep — COMMENTED** [review]
Cool

File: src/test/kotlin/coverosR3z/timerecording/CreateEmployeeAPITests.kt [hunk 1]
```diff
@@ -110,4 +77,93 @@ class CreateEmployeeAPITests {
...
+        assertEquals(StatusCode.OK, CreateEmployeeAPI.handleGet(sd).statusCode)
+    }
+
+    /**
+     * Why would the system be GET'ing this page? disallow
+     */
+    @Category(APITestCategory::class)
+    @Test
+    fun testShouldDisallowSystemToGetPage() {
+        val sd = makeServerData(PostBodyData(), tru, au, user = SYSTEM_USER)
```
Comment (NihilistOwl) [inline src/test/kotlin/coverosR3z/timerecording/CreateEmployeeAPITests.kt:139]: Isn't user = SYSTEM_USER the default? Also, could we refactor some of these in the same way as LoggingAPITests?

Comment (7ep) [reply src/test/kotlin/coverosR3z/timerecording/CreateEmployeeAPITests.kt:139]: yes and yes, and on using SYSTEM_USER, it's just to make it more explicit and legible when reading the test.

**Pull request merged**